x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -139,4 +139,14 @@ public void testDateRangeCast() {
         assertEquals("date", rq.field());
         assertEquals(DateTime.parse("1969-05-13T12:34:56Z"), rq.lower());
     }
+    
+    public void testLikeConstructsNotSupported() {
+        LogicalPlan p = plan("SELECT LTRIM(keyword) lt FROM test WHERE LTRIM(keyword) LIKE '%a%'");
+        assertTrue(p instanceof Project);
+        p = ((Project) p).child();
+        assertTrue(p instanceof Filter);
+        Expression condition = ((Filter) p).condition();
+        SqlIllegalArgumentException ex = expectThrows(SqlIllegalArgumentException.class, () -> QueryTranslator.toQuery(condition, false));
+        assertEquals("Scalar functions (LTRIM(keyword)) not allowed (yet) as arguments for LIKE", ex.getMessage());
```

Comment:
Minor nitpick, the plural in `functions` should be optional (it's easy to check the number of arguments and add it only if needed). Also the function is `LTRIM` not `LTRIM(keyword)` - namely functionName() not function().
Lastly I would pick a different function to check whether the proper name is returned (say `CHAR_LENGTH`) without the casing.